src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -197,90 +216,165 @@ public boolean isPageUri(String uri) {
+     * @param locator used to find the element
+     * @see AppPage#waitForElementStaleness(WebElement)
+     */
+    public void waitForElementStaleness(By locator) {
+        waitForElementStaleness(browser.driver.findElement(locator));
+    }
+
+    /**
+     * Wait until an element is no longer attached to the DOM or the timeout expires.
+     * @param element the WebElement
+     * @throws org.openqa.selenium.TimeoutException if the timeout defined in
+     * {@link TestProperties#TEST_TIMEOUT} expires
+     * @see org.openqa.selenium.support.ui.FluentWait#until(com.google.common.base.Function)
+     */
+    public void waitForElementStaleness(WebElement element) {
+        waitFor(ExpectedConditions.stalenessOf(element));
+    }
+
+    /**
+     * Wait until an element belongs to the class or the timeout expires.
+     * @param element the WebElement
+     * @param elementClass the class that the element must belong to
+     * @throws org.openqa.selenium.TimeoutException if the timeout defined in
+     * {@link TestProperties#TEST_TIMEOUT} expires
+     * @see org.openqa.selenium.support.ui.FluentWait#until(com.google.common.base.Function)
+     */
+    void waitForElementToBeMemberOfClass(WebElement element, String elementClass) {
+        waitFor(driver -> {
+            String classAttribute = element.getAttribute("class");
+            List<String> classes = Arrays.asList(classAttribute.split(" "));
+
+            return classes.contains(elementClass);
+        });
     }
 
     /**
      * Waits for element to be invisible or not present, or timeout.
      */
     public void waitForElementToDisappear(By by) {
-        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.TEST_TIMEOUT);
-        wait.until(ExpectedConditions.invisibilityOfElementLocated(by));
+        waitFor(ExpectedConditions.invisibilityOfElementLocated(by));
     }
 
     /**
      * Waits for a list of elements to be invisible or not present, or timeout.
      */
     public void waitForElementsToDisappear(List<WebElement> elements) {
-        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.TEST_TIMEOUT);
-        wait.until(ExpectedConditions.invisibilityOfAllElements(elements));
+        waitFor(ExpectedConditions.invisibilityOfAllElements(elements));
     }
 
     /**
-     * Waits for an alert to appear on the page, up to the timeout specified.
+     * Waits until there is no scrolling on the page. Note: The detection of whether the page is scrolling is coupled to our
+     * own implementation of scrolling {@code (scrollTo.js)} and is not a true detection.
+     * @throws org.openqa.selenium.TimeoutException if the timeout defined in {@link TestProperties#TEST_TIMEOUT} expires
      */
-    public void waitForAlertPresence() {
-        WebDriverWait wait = new WebDriverWait(browser.driver, TestProperties.TEST_TIMEOUT);
-        wait.until(ExpectedConditions.alertIsPresent());
+    void waitForNoScrolling() {
```

Comment:
I think I originally named it something like that but went with `waitForNoScrolling` because the method does not actually wait for scrolling to complete. Rather it waits for the page to "freeze" which means if there is no scrolling to begin with the method will instantly finish waiting.

`waitForNoScrolling` sounds awkward though but I'm not sure how to better name it.

Reply:
Well, that can be said for all the wait methods - they return immediately if the condition was true to begin with.

Reply:
If the page is not scrolling, it can be said that scrolling is complete.

Reply:
I guess from another point of view that works too.